io/io/inc/TFile.h

```diff
@@ -300,6 +301,9 @@ class TFile : public TDirectoryFile {
    static TFile       *Open(const char *name, Option_t *option = "",
                             const char *ftitle = "", Int_t compress = ROOT::RCompressionSetting::EDefaults::kUseCompiledDefault,
                             Int_t netopt = 0);
+   static std::unique_ptr<TFile> MakeUnique(const char *name, Option_t *option = "",
```

Comment:
I would intuitively expect to have `TFile::MakeUnique` behave the same as
```
std::make_unique<TFile>{ arguments };
```
i.e. only work on local file.
What about:
```suggestion
   static std::unique_ptr<TFile> OpenUniquePtr(const char *name, Option_t *option = "",
```
or similar?

Reply:
`OpenUnique` also has the "tab-completion advantage", users will more easily find it in docs etc.

Reply:
I had called this function that way originally, then the `MakeUnique` naming was proposed. I have no strong preference here. I see your point, I would like to hear also other opinions

Reply:
My personal pros/cons:
- `TFile::OpenUniquePtr()`: we're not opening a `unique_ptr`
- `TFile::OpenUnique()`: sounds like an exclusively owned file, where no other client can open that file.
- `TFile::MakeUnique()`: follows a known pattern!

The fact that `make_unique<TFile>` does something else is the fault of the public `TFile` ctor ;-)

Independently from this PR @eguiraud proposed that `TFile::TFile()` should complain ("use TFile::Open()!") if it detects "://" in the filename. That might alleviate some of your concern, @pcanal